File: 2021/01/31/beomjo/algorithm/BOJ/java/17413.java

import java.io.*;
import java.util.Stack;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        String input = br.readLine() + "\n";

        Stack<Character> stack = new Stack<>();

        boolean isTag = false;

        for (int i = 0; i < input.length(); i++) {
            char c = input.charAt(i);

            if (c == '<') isTag = true;
            if (isTag || c == ' ' || c == '\n') {
                while (!stack.isEmpty()) {
                    bw.write(stack.pop());
                }
                if (c != '\n') bw.write(c);
            } else {
                stack.push(c);
            }
            if (c == '>') isTag = false;
        }
        bw.flush();
        return;
    }
}

File: 2021/01/31/beomjo/algorithm/BOJ/kotlin/17413.kt
import java.lang.StringBuilder
import java.util.*

fun main() {
    val stack = Stack<Char>()
    val sb = StringBuilder()
    var isTag = false
    "${readLine()!!}\n".forEach { c ->
        if (c == '<') isTag = true
        if (isTag || c == ' ' || c == '\n') {
            while (!stack.isEmpty()) {
                sb.append(stack.pop())
            }
            if (c != '\n') sb.append(c)
        } else {
            stack.push(c);
        }
        if (c == '>') isTag = false
    }
    print(sb)
}